components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -108,11 +110,29 @@
 
     private static final Log log = LogFactory.getLog(DefaultIDTokenBuilder.class);
     private JWSAlgorithm signatureAlgorithm = null;
+    private JWEAlgorithm encryptionAlgorithm;
+    private EncryptionMethod encryptionMethod;
 
     public DefaultIDTokenBuilder() throws IdentityOAuth2Exception {
         //map signature algorithm from identity.xml to nimbus format, this is a one time configuration
         signatureAlgorithm = OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(
                 OAuthServerConfiguration.getInstance().getIdTokenSignatureAlgorithm());
+        encryptionAlgorithm = OAuth2Util.mapEncryptionAlgorithmForJWEAlgorithm(
```

Comment:
As an improvement we are going to allow configuring encryption algorithm per Service Provider. So we shouldn't be having this mapping inside the constructor since its going to vary per each service provider